From 3ff889b8b3029142b2e05bfdbaf253596fd3ad7c Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 1 Dec 2021 18:27:06 +0800
Subject: [PATCH 15/17] blink: rtc: Force disabling optimized video scaling

This would break VEA when encoder requires special coded size:
https://bugs.chromium.org/p/chromium/issues/detail?id=1193457#c6

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../renderer/platform/peerconnection/rtc_video_encoder.cc     | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/third_party/blink/renderer/platform/peerconnection/rtc_video_encoder.cc b/third_party/blink/renderer/platform/peerconnection/rtc_video_encoder.cc
index 14d2a2f5..4604982b 100644
--- a/third_party/blink/renderer/platform/peerconnection/rtc_video_encoder.cc
+++ b/third_party/blink/renderer/platform/peerconnection/rtc_video_encoder.cc
@@ -1218,6 +1218,10 @@ void RTCVideoEncoder::Impl::EncodeOneFrame() {
 #else
         false;
 #endif
+
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=1193457#c6
+    optimized_scaling = false;
+
     if (optimized_scaling) {
       DCHECK_EQ(buffer->type(), webrtc::VideoFrameBuffer::Type::kNative);
       auto scaled_buffer = buffer->Scale(input_visible_size_.width(),
-- 
2.17.1

